projects/m5/001-budget-app/js/index.js

```diff
@@ -0,0 +1,225 @@
+        let titleLine =  '' //****category name*****/
+        let ledgerString = []
+
+        //wirte the first line of "*"
+        for (let i = 1; i <= (maxCharNumber-this.name.length);i++){
+            titleLine += '*'
+        }
+        //concatenation of string (* until where to start category name + category name + * other half)
+        titleLine = titleLine.substring(0,halfTitle) + this.name + titleLine.substring(halfTitle)
+
+        //cicle in the ledger list
+        this.ledger.forEach(element => {
+            /* print the first 23 char of the description, add the quantity that is written starting after the 23 char
+             * in this way is going alway to be align on the rigth
+             */
+            ledgerString.push((element['description'].substring(0,23) + 
+            (element['amount'].toFixed(2).toString().padStart((maxCharNumber-element['description'].slice(0,23).length)))))
+        })
+        return console.log(`${titleLine}\n${ledgerString.join('\n')}\nTotal: ${this.getBalance()}`)
+    }
+}
+
+/**
+ * Function to create a spend chart
+ * @param {string} listOfCategory string of category that have been already initialized
+ */
+const createSpendChart = (listOfCategory) => {
+    
+    listOfCategory.length = 4 // initilized the max number of the argument that the function can accept in the list
+    let categoryNames = [] //list of the argument .name of the choosen category
+    let elementPercentage = [] // list of total withdraws for every category {number}
+    let totalElementPercentage = [] // list of withdraw for every category transformed in % {number}
+    let totalCategoryWithdraw = 0 // total ammount of withdraw for every category (change everytime the category change in the loop)
+    let totalWithdraw = 0 // total amount of all widhdraw of all category together
+
+    /**
+     * foreach to go in each category choosen
+     */
+    listOfCategory.forEach(category => {
+
+        categoryNames.push(category.name)
+
+        //if the category choosen are less than 4, but the length is initialized at 4, then there will be some undefined element in the array
+        if(category != undefined){
+            category.ledger.forEach(singleAmount =>{
+                if (singleAmount['amount'] < 0) { //to consider only withdraw because are negtive number
+                    totalWithdraw = (+(totalWithdraw - (-singleAmount['amount'])).toFixed(2)) 
+                    totalCategoryWithdraw = (+(totalCategoryWithdraw - (-singleAmount['amount'])).toFixed(2))
+                }
+            })  
+        }
+        elementPercentage.push(Math.abs(totalCategoryWithdraw))
+        totalCategoryWithdraw = 0 //next category start from 0
+    })
+    
+    totalWithdraw = Math.abs(totalWithdraw)
+    totalElementPercentage = elementPercentage.map(x => parseInt(((x/totalWithdraw)*100)/10,10)*10) //list rounded at the 10 smaller
```

Comment:
No best practice. Declare directly const here.
Delete row 108.
To Fix other similar errors.

Reply:
ok.